routes/apiRoutes.js

```diff
@@ -1,33 +1,103 @@
-// const db = require("../models");
-// const axios = require("axios");
+const db = require('../models/index')
+const axios = require('axios')
 
-// Get users
-// Get "needed posts"
+module.exports = function (app) {
+  // ==================================================================================================
+  // Create "need" via TILE(images of preset groceries) or INPUT BOX(custom need, like medicine)
+  // ==================================================================================================
+  app.post('/api/need', async (req, res) => {
+    // Save Need to Database
+    const request = await db.Needs.create({
+      // name: req.body.name,
+      // description: req.body.description,
+      // completed: req.body.completed,
+      // image: req.body.image,
+      // user: req.body.user
+      name: 'Tuna Fish',
+      description: 'Description',
+      completed: false,
+      image: 'image',
+      user: 'Kurt'
+    })
+    // Send the request back to the front end
+    res.send(request)
+  })
 
-// Post "needed post/tile?"
+  // ==================================================================================================
+  // Get all "needs" from local database based location
+  // ==================================================================================================
+  app.get('/api/needs', async (req, res) => {
+    const request = await db.Needs.find({})
+    // Send the request back to the front end
+    res.send(request)
+  })
 
-// Delete "postings"
+  // ==================================================================================================
+  // Get single "need" from local database based on the user
+  // ==================================================================================================
+  app.get('/api/need/:id', async (req, res) => {
+    const request = await db.Needs.find({ users: { _id: req.params.id } })
+    // Send the request back to the front end
+    res.send(request)
+  })
 
-// Update "postings" to complete/incomplete
+  // ==================================================================================================
+  // Update "need" to COMPLETE
```

Comment:
'to complete' implies you're only able to complete a  need through this api endpoint. is that the intention?